File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/SpikeSlime_S_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class SpikeSlime_S_Model extends MonsterInfo{
    public static Id = "SpikeSlime_S"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(11, 15);
        else
            return Mathf.RandomRange(10,14);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/slimeAltS/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/slimeAltS/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/slimeAltS/skeleton.png"));
    }

    get Name(): string {
        return "尖刺史莱姆（小）";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/SpikeSlime_S_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class SpikeSlime_S_Model : MonsterInfo
    {
        public static string Id = "SpikeSlime_S";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(11, 15) : Random.Range(10, 14);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/slimeAltM/skeleton.atlas",
                    "Monsters/theBottom/slimeAltM/skeleton.json", "Monsters/theBottom/slimeAltM/skeleton.png");
            }
        }

        public override string Name => "尖刺史莱姆（小）";

        public SpikeSlime_S_Model(Vector2 pos) : base(pos)
        {
        }
    }
}